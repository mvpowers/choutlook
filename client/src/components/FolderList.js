import React from 'react';
import styled from 'styled-components';
import {
  FaInbox,
  FaPencil,
  FaListAlt,
  FaBan,
  FaArrowRight,
  FaTrashO,
} from 'react-icons/lib/fa';
import { Folder, FolderSection, FolderLink } from './';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  background-color: #f4f4f4;
  color: #5084aa;
  border-left: 1px solid #eaeaea;
  overflow-y: hidden;
`;

export default () => (
  <Container>
    <FolderSection name="Favorites" />
    <Folder name="Inbox" icon={<FaInbox size={16} />} />
    <Folder name="Drafts" icon={<FaPencil size={16} />} />
    <Folder name="Archive" icon={<FaListAlt size={16} />} />
    <FolderLink name="Add favorite" />
    <FolderSection name="Folders" />
    <Folder name="Inbox" icon={<FaInbox size={16} />} />
    <Folder name="Junk Email" icon={<FaBan size={16} />} />
    <Folder name="Drafts" icon={<FaPencil size={16} />} />
    <Folder name="Sent Items" icon={<FaArrowRight size={16} />} />
    <Folder name="Deleted" icon={<FaTrashO size={16} />} />
    <Folder name="Archive" icon={<FaListAlt size={16} />} />
    <FolderLink name="New folder" />
  </Container>
);
