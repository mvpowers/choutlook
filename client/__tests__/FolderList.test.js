import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FolderList from '../src/components/FolderList';

Enzyme.configure({ adapter: new Adapter() });

describe('Folder', () => {
  it('should be defined', () => {
    expect(FolderList).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = shallow(<FolderList />);
    expect(tree).toMatchSnapshot();
  });
});
