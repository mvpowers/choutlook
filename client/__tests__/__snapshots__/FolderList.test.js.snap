// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Folder should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <FolderSection
          name="Favorites"
        />,
        <Folder
          icon={
            <FaInbox
              size={16}
            />
          }
          name="Inbox"
        />,
        <Folder
          icon={
            <FaPencil
              size={16}
            />
          }
          name="Drafts"
        />,
        <Folder
          icon={
            <FaListAlt
              size={16}
            />
          }
          name="Archive"
        />,
        <FolderLink
          name="Add favorite"
        />,
        <FolderSection
          name="Folders"
        />,
        <Folder
          icon={
            <FaInbox
              size={16}
            />
          }
          name="Inbox"
        />,
        <Folder
          icon={
            <FaBan
              size={16}
            />
          }
          name="Junk Email"
        />,
        <Folder
          icon={
            <FaPencil
              size={16}
            />
          }
          name="Drafts"
        />,
        <Folder
          icon={
            <FaArrowRight
              size={16}
            />
          }
          name="Sent Items"
        />,
        <Folder
          icon={
            <FaTrashO
              size={16}
            />
          }
          name="Deleted"
        />,
        <Folder
          icon={
            <FaListAlt
              size={16}
            />
          }
          name="Archive"
        />,
        <FolderLink
          name="New folder"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "Favorites",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": <FaInbox
            size={16}
          />,
          "name": "Inbox",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": <FaPencil
            size={16}
          />,
          "name": "Drafts",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": <FaListAlt
            size={16}
          />,
          "name": "Archive",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "Add favorite",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "Folders",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": <FaInbox
            size={16}
          />,
          "name": "Inbox",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": <FaBan
            size={16}
          />,
          "name": "Junk Email",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": <FaPencil
            size={16}
          />,
          "name": "Drafts",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": <FaArrowRight
            size={16}
          />,
          "name": "Sent Items",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": <FaTrashO
            size={16}
          />,
          "name": "Deleted",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": <FaListAlt
            size={16}
          />,
          "name": "Archive",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "New folder",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <FolderSection
            name="Favorites"
          />,
          <Folder
            icon={
              <FaInbox
                size={16}
              />
            }
            name="Inbox"
          />,
          <Folder
            icon={
              <FaPencil
                size={16}
              />
            }
            name="Drafts"
          />,
          <Folder
            icon={
              <FaListAlt
                size={16}
              />
            }
            name="Archive"
          />,
          <FolderLink
            name="Add favorite"
          />,
          <FolderSection
            name="Folders"
          />,
          <Folder
            icon={
              <FaInbox
                size={16}
              />
            }
            name="Inbox"
          />,
          <Folder
            icon={
              <FaBan
                size={16}
              />
            }
            name="Junk Email"
          />,
          <Folder
            icon={
              <FaPencil
                size={16}
              />
            }
            name="Drafts"
          />,
          <Folder
            icon={
              <FaArrowRight
                size={16}
              />
            }
            name="Sent Items"
          />,
          <Folder
            icon={
              <FaTrashO
                size={16}
              />
            }
            name="Deleted"
          />,
          <Folder
            icon={
              <FaListAlt
                size={16}
              />
            }
            name="Archive"
          />,
          <FolderLink
            name="New folder"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "Favorites",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": <FaInbox
              size={16}
            />,
            "name": "Inbox",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": <FaPencil
              size={16}
            />,
            "name": "Drafts",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": <FaListAlt
              size={16}
            />,
            "name": "Archive",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "Add favorite",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "Folders",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": <FaInbox
              size={16}
            />,
            "name": "Inbox",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": <FaBan
              size={16}
            />,
            "name": "Junk Email",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": <FaPencil
              size={16}
            />,
            "name": "Drafts",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": <FaArrowRight
              size={16}
            />,
            "name": "Sent Items",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": <FaTrashO
              size={16}
            />,
            "name": "Deleted",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": <FaListAlt
              size={16}
            />,
            "name": "Archive",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "New folder",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
